build-tools/src/main/java/org/elasticsearch/gradle/plugin/PluginBuildPlugin.java

```diff
@@ -51,6 +55,27 @@ public void apply(final Project project) {
             task.getOutputFile().set(file);
         });
 
-    }
+        project.getTasks().withType(GenerateTestBuildInfoTask.class).named("generateTestBuildInfo").configure(task -> {
+            var jarTask = project.getTasks().withType(Jar.class).named("jar").get();
+            String moduleName = (String) jarTask.getManifest().getAttributes().get("Automatic-Module-Name");
+            if (moduleName == null) {
+                moduleName = jarTask.getArchiveBaseName().getOrNull();
+            }
+            if (moduleName != null) {
+                task.getModuleName().set(moduleName);
+            }
+            var propertiesExtension = project.getExtensions().getByType(PluginPropertiesExtension.class);
+            task.getComponentName().set(providerFactory.provider(propertiesExtension::getName));
+            task.getOutputFileName().set("plugin-test-build-info.json");
+        });
 
+        project.getTasks().withType(ProcessResources.class).named("processResources").configure(task -> {
+            var componentName = project.getExtensions().getByType(PluginPropertiesExtension.class).getName();
+            var pluginProperties = project.getTasks().withType(GeneratePluginPropertiesTask.class).named("pluginProperties");
+            task.into("META-INF/es-plugins/" + componentName + "/", copy -> {
```

Comment:
I'm wondering if we're going to hit ordering issues with `componentName` here. That is, Gradle resolves this plugin configuraiton before the name actually get's configured in the build script. This order is hard to guarantee, which is why we try to use lazy `Property` everywhere. One way to avoid this is to pass a `Callable` which returns this to `task.into()` for the destination location so that gets resolved at execution time.

Reply:
I wrapped the required values in Callable for both `componentName` and `pluginProperties`. It seems to work. For the last value for getting the path the entitlement-file, I assumed this was fine to leave it as is since the location of the file already exists always.